templates.rst

```diff
@@ -0,0 +1,1061 @@
+the new URL.
+
+Consider the following routing configuration:
+
+.. configuration-block::
+
+    .. code-block:: php-annotations
+
+        // src/Controller/BlogController.php
+
+        // ...
+        use Symfony\Component\Routing\Annotation\Route;
+
+        class BlogController extends AbstractController
+        {
+            /**
+             * @Route("/", name="blog_index")
+             */
+            public function index()
+            {
+                // ...
+            }
+
+            /**
+             * @Route("/article/{slug}", name="blog_post")
+             */
+            public function show(string $slug)
+            {
+                // ...
+            }
+        }
+
+    .. code-block:: yaml
+        # config/routes.yaml
+        blog_index:
+            path:       /
+            controller: App\Controller\BlogController::index
+
+        blog_post:
+            path:       /article/{slug}
+            controller: App\Controller\BlogController::show
+
+    .. code-block:: xml
+
+        <!-- config/routes.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <routes xmlns="http://symfony.com/schema/routing"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/routing
+                https://symfony.com/schema/routing/routing-1.0.xsd">
+
+            <route id="blog_index"
+                path="/"
+                controller="App\Controller\BlogController::index"/>
+
+            <route id="blog_post"
+                path="/article/{slug}"
+                controller="App\Controller\BlogController::show"/>
+        </routes>
+
+    .. code-block:: php
+
+        // config/routes.php
+        use App\Controller\BlogController;
+        use Symfony\Component\Routing\Loader\Configurator\RoutingConfigurator;
+
+        return function (RoutingConfigurator $routes) {
+            $routes->add('blog_index', '/')
+                ->controller([BlogController::class, 'index'])
+            ;
+
+            $routes->add('blog_post', '/articles/{slug}')
+                ->controller([BlogController::class, 'show'])
+            ;
+        };
+
+Use the ``path()`` Twig function to link to these pages and pass the route name
+as the first argument and the route parameters as the optional second argument:
+
+.. code-block:: html+twig
+
+    <a href="{{ path('blog_index') }}">Homepage</a>
+
+    {# ... #}
+
+    {% for post in blog_posts %}
+        <h1>
+            <a href="{{ path('blog_post', {slug: post.slug}) }}">{{ post.title }}</a>
+        </h1>
+
+        <p>{{ post.excerpt }}</p>
+    {% endfor %}
+
+The ``path()`` function generates relative URLs. If you need to generate
+absolute URLs (for example when rendering templates for emails or RSS feeds),
+use the ``url()`` function, which takes the same arguments as ``path()``
+(e.g. ``<a href="{{ url('blog_index') }}"> ... </a>``).
```

Comment:
This entire section is repeated in the routing chapter. https://symfony.com/doc/current/routing.html#generating-urls-in-templates

Probably we should keep this new section here. Then, in routing, we keep the `Generating URLs in Templates` (so it's findable), but link here.